src/date.rs

```diff
@@ -276,21 +276,16 @@ impl<Tz: TimeZone> Date<Tz> {
         self.date
     }
 
-    /// Retrieves the elapsed years from now to the given [`Date`].
-    #[cfg(feature = "clock")]
-    pub fn elapsed_years(&self) -> u32 {
-        let now = Utc::today().with_timezone(&self.timezone());
-
-        let years = if (now.month(), now.day()) < (self.month(), self.day()) {
-            now.year() - self.year() - 1
-        } else {
-            now.year() - self.year()
-        };
-
-        if years.is_positive() {
-            years as u32
-        } else {
-            0
+    /// Returns the number of whole years from the given `base` until `self`.
+    pub fn years_since(&self, base: Self) -> Option<u32> {
+        let mut years = self.year() - base.year();
+        if (self.month(), self.day()) < (base.month(), base.day()) {
+            years -= 1;
+        }
+
+        match years >= 0 {
+            true => Some(years as u32),
+            false => None,
```

Comment:
That's a sophisticated if-else xD

Is it intentional? It seems you did that everywhere. Maybe to avoid if-expressions?

Reply:
I find `match` expressions easier to read (in the case where the branch fits on one line): they're slightly more vertically compact and they lay out both branches right next to each other. I use these all the time, so I wouldn't describe this as particularly sophisticated.

Reply:
Yes as I never do that it looks pretty strange to me but I think you have a good point

Reply:
Personally I prefer to use functions like `is_positive()` instead of `>= 0`.

Reply:
It's not obvious to me without actually testing it what `is_positive()` returns for `0`. The direct comparison is also more compact. Why do you prefer the method call here?

Reply:
For the readability (but it's a personal habits). `is_positive()` returns `false` if the number is either zero or negative.